src/TracedRNumber.jl

```diff
@@ -12,6 +12,9 @@ mutable struct TracedRNumber{T} <: RNumber{T}
     end
 end
 
+get_mlir_data(x::TracedRNumber) = x.mlir_data
+set_mlir_data!(x::TracedRNumber, data) = (x.mlir_data = data)
```

Comment:
would you mind making `set_mlir_data!` return `x` instead of `data`? this way some other implementations become straightforward

```suggestion
set_mlir_data!(x::TracedRNumber, data) = (x.mlir_data = data; return x)
```